packages/backend/src/app.ts

```diff
@@ -577,7 +577,7 @@ export class App {
       createTokenIntrospectionMiddleware({
         requestType: AccessType.IncomingPayment,
         requestAction: RequestAction.Read,
-        bypassError: true
+        canSkipAuthValidation: true
       }),
       authenticatedStatusMiddleware,
```

Comment:
In `authenticatedStatusMiddleware` we should return early if we know that we ended up skipping auth validation in `createTokenIntrospectionMiddleware` (since it doesn't make sense to do a lot of work for something we already know shouldn't succeed). One way we can do this is to make `client` optional in `HttpSigWithAuthenticatedStatusContext`, and only proceed to call `throwIfSignatureInvalid` if `ctx.client` is defined (setting `ctx.client` is the main function of `createTokenIntrospectionMiddleware`)

Reply:
what do you think about checking for 
`if (!ctx.request.headers.authorization) {
    await next()
    return
  }`
instead of making the `client' optional (string | undefined) for the rest of the code? I also see that `authenticatedStatusMiddleware` is only used for open payments